### DISCUSSION TITLE
GitHub GraphQL API: `commit.statusCheckRollup.state` is returning as `null`

### DISCUSSION THREAD

**Discussion by BravoNatalie** [community/community#38599] [Apps, API and Webhooks] [answered]
I'm trying to run an action that uses the graphql API, but I'm not being able to check the `statusCheckRollup.state`.

This is the query I'm using on the GraphQL API Explorer (but also in my own code):

```
 query($owner: String!, $repo: String!, $pull_number: Int!) {
    repository(owner: $owner, name: $repo) {
      pullRequest(number:$pull_number) {
        commits(last: 1) {
          nodes {
            commit {
              statusCheckRollup {
                state
              }
            }
          }
        }
      }
    }
  }
```
The `statusCheckRollup.state` is always null, even though the GraphQL API Explorer has the right access.

What is wrong? I have used this query before and it worked.

**Comment by david-engelmann** [comment]
Did you pass the variables correctly?I was able to run the query successfully, see my screenshot below.

![image](https://user-images.githubusercontent.com/81196676/200671679-2ae5e28e-4138-4e02-9891-1fe332f1aad6.png)

**Reply by BravoNatalie** [reply]
yes, everything is correct

check the example:
<img width="1107" alt="image" src="https://user-images.githubusercontent.com/37748828/200681692-d68368ad-d91c-481a-b440-9da284f47953.png">

**Comment by david-engelmann** [comment, marked as answer]
I'm not sure if the repo in question will have a statusCheckRollup if an action isn't associated with the commit, see the attach screenshot

![image](https://user-images.githubusercontent.com/81196676/200692758-4691729a-72e7-47a5-bb8d-ac659bea424f.png)

[I added an update to the workflow trigger to be caused by a PR to the master branch](https://github.com/david-engelmann/Proffy/pull/1). After that the query worked as expected.

![image](https://user-images.githubusercontent.com/81196676/200693683-7e9f6d75-a3c0-41e8-b254-be9e93226f5f.png)

**Reply by BravoNatalie** [reply]
hmm, I see.  Is there a way to see if I can merge the PR without any problems if there's no action associated?

**Reply by BravoNatalie** [reply]
just saw that I can use this in the case of not having a `statusCheckRollup`:

```
query($owner: String!, $repo: String!, $pull_number: Int!) {
  repository(owner: $owner, name: $repo) {
    pullRequest(number:$pull_number) {
      mergeable
    }
  }
}
```

**Reply by david-engelmann** [reply]
Perfect! Feel free to mark it as answered!

**Reply by LangLangBart** [reply]
> hmm, I see. Is there a way to see if I can merge the PR without any problems if there's no action associated?

### schema previews - If you need more information besides `mergeable `
The `GraphQL` schema can be expanded with an additional Header to get `PullRequest.canBeRebased` and `PullRequest.mergeStateStatus`.

Source: [GitHub Docs GraphQL schema - Merge info preview](https://docs.github.com/en/graphql/overview/schema-previews#merge-info-preview)
<img src="https://user-images.githubusercontent.com/92653266/200754950-e4c67c1b-8bb6-4660-b2d0-6164e6a838d7.png" width="600">

### example
```graphql
{
  repository(owner: "cli", name: "cli") {
    pullRequest(number: 6578) {
      mergeStateStatus
      canBeRebased
      mergeable
      state
      commits(last: 1) {
          nodes {
            commit {
              statusCheckRollup {
                state
              }
            }
          }
        }
    }
  }
}
```
```json
{
  "data": {
    "repository": {
      "pullRequest": {
        "mergeStateStatus": "BLOCKED",
        "canBeRebased": true,
        "mergeable": "MERGEABLE",
        "state": "OPEN",
        "commits": {
          "nodes": [
            {
              "commit": {
                "statusCheckRollup": {
                  "state": "SUCCESS"
                }
              }
            }
          ]
        }
      }
    }
  }
}
```
